 <launch>

    <arg name="arm_base_link" value="arm5/kinematic_base"/>

    <arg name="arm_aruco_joint" value="arm5/end_effector"/>
    <arg name="arm_aruco_frame" value="end_effector_marker2"/>

    <arg name="camera_frame" value="stereo"/>

    <!--arg name="sense3d_base" value="sense3d_base"/>
    <arg name="girona500" value="girona500"/-->
    <!--LASER node name="static_tf_sense3d" type="static_transform_publisher" pkg="tf" args="-0.296713 -0.147335 0.20218 -1.60815 0.188877 -0.272845 $(arg arm_base_link) $(arg sense3d_base) 100"/-->
    <!--GIRONA node name="static_tf_girona500_arm" type="static_transform_publisher" pkg="tf" args="0.18052 -0.0275 0.66 3.1416 0 0 $(arg girona500) $(arg arm_base_link) 100"/-->
    <!--BUMBLEBEE node name="static_tf_stereo" type="static_transform_publisher" pkg="tf" args="-0.356612 -0.056596 0.119828 1.50441 -0.0300195 0.511419 $(arg arm_base_link) stereo2 100"/-->
    <!-- STEREO FROM BB node name="static_tf_stereo_frombb" type="static_transform_publisher" pkg="tf" args="-0.356612 -0.045 0.119828 1.57 0 0.511419 $(arg arm_base_link) stereo2 100"/>
    <node name="static_tf_stereo2" type="static_transform_publisher" pkg="tf" args="0.09 0 0 3.1416 0 0 stereo2 $(arg camera_frame) 100"/-->


    <!--node name="tf_stereo_from_bumblebe" type="static_transform_publisher" pkg="tf" args="-0.356612 0.045 0.119828 -1.57 0 -0.511419 $(arg arm_base_link) stereo 100"/-->

    <node name="tf_stereo_from_solid" type="static_transform_publisher" pkg="tf" args="-0.29465 0.045 0.11951 -1.57 0 -0.4625123 $(arg arm_base_link) stereo 100"/>

    <include file="$(find pm_manipulation)/launch/joint_offset.launch"/>

     <param name="robot_description" textfile="$(find pm_manipulation)/launch/arm5e_with_sensor.urdf"/>
     <node pkg="robot_state_publisher" type="state_publisher" name="rob_st_pub" >
      <remap from="joint_states" to="/arm5e/joint_state_angle_fixed" />
      <param name="publish_frequency" type="double" value="100.0" />
      <param name="tf_prefix" type="string" value="arm5" /> 
    </node>

    <!--node name="static_tf_aruco_joint_old" type="static_transform_publisher" pkg="tf" args="-0.00984765 0.0649946 -0.10452 -0.0612567 0.170656 -1.83375 $(arg arm_aruco_joint) $(arg arm_aruco_frame) 10"/-->
    

  </launch>
